### ISSUE TITLE
Incorrect indentation for multi line

### ISSUE THREAD

**Issue by dsvua** [adrienverge/yamllint#365]
I have this in yaml used by kustomize:
```
---
patches:
  - target:
      kind: RunnerDeployment
      name: .*
    patch: |-
      - op: replace
        path: /metadata/name
        value: chef-repo-runner

```
It complains about identation for `path` and `value` lines.
```
[indentation] wrong indentation: expected 6 but found 8
```

Is it a bug or I am doing something wrong.

**Comment by dsvua** [comment]
This is my .yamllint.yaml:
```
---

extends: default

rules:
  # 80 chars should be enough, but don't fail if a line is longer
  line-length:
    max: 80
    level: warning

  indentation:
    spaces: 2
    # indent-sequences: consistent
    check-multi-line-strings: true
    level: error
  key-duplicates:
    level: error
  trailing-spaces:
    level: error
  new-line-at-end-of-file:
    level: error
  brackets:
    level: error
  empty-lines:
    level: warning
  document-start:
    present: true
    level: error
```

**Comment by adrienverge** [comment]
This is intended because your configuration enables `indentation: {check-multi-line-strings: true}`. 

The YAML spec states that the following part must be interpreted as a string, because it's declared using `|-`.
```
- op: replace
  path: /metadata/name
  value: chef-repo-runner
```
Hence, the 2 last lines are reported, exactly like if they were plain text:

```
line 1
  line 2
  line 3
```
